.github/renovate.json

```diff
@@ -122,5 +122,42 @@
       "matchPackageNames": ["@balena/open-balena-api"],
       "matchUpdateTypes": ["minor"]
     }
+  ],
+  "regexManagers": [
+    {
+      "fileMatch": [
+        "^Dockerfile$",
+        "(^|/)Dockerfile[^/]*$"
+      ],
+      "matchStrings": [
+        "ENV NODE_VERSION=(?<currentValue>.*?)\\n",
+        "ARG NODE_VERSION=(?<currentValue>.*?)\\n"
+      ],
+      "depNameTemplate": "node",
+      "datasourceTemplate": "node"
+    },
+    {
+      "fileMatch": [
+        "^Dockerfile$",
+        "(^|/)Dockerfile[^/]*$"
+      ],
+      "matchStrings": [
+        "ENV NPM_VERSION=(?<currentValue>.*?)\\n",
+        "ARG NPM_VERSION=(?<currentValue>.*?)\\n"
+      ],
+      "depNameTemplate": "npm",
+      "datasourceTemplate": "npm"
+    },
+    {
+      "fileMatch": [
+        "(^|/)(?:docker-)?compose[^/]*\.ya?ml$",
+        "(^|/)Dockerfile[^/]*$"
```

Comment:
@ab77, this regex matches everything that contains "Dockerfile" within it e.g., Dockerfiledsafda. Is this something we want?

Reply:
@olamiko we could tune the second regex to only match `Dockerfile.*` and `Dockerfile`.

Reply:
@olamiko @ab77 I think we should just use `"(^|/)Dockerfile(\.[:alnum:]+)?$"` and drop `"(^|/)Dockerfile$"`

Dockerfile
Dockerfile.template
Dockerfile.foobar

JF thread (internal only): https://jel.ly.fish/thread-d2ebed26-cb33-4b4c-93a9-86baddd47e17